scadalts-ui/src/views/Reports/ReportsPage.vue

```diff
@@ -0,0 +1,941 @@
+			this.reportForm.fromYear= this.selectedReport.fromYear; 
+            this.reportForm.fromMonth= this.selectedReport.fromMonth; 
+			this.reportForm.fromDay= this.selectedReport.fromDay; 
+			this.reportForm.fromHour= this.selectedReport.fromHour; 
+			this.reportForm.fromMinute= this.selectedReport.fromMinute; 
+			this.reportForm.toNone= this.selectedReport.toNone; 
+			this.reportForm.toYear= this.selectedReport.toYear; 
+			this.reportForm.toMonth= this.selectedReport.toMonth; 
+            this.reportForm.toDay= this.selectedReport.toDay; 
+			this.reportForm.toHour= this.selectedReport.toHour; 
+			this.reportForm.toMinute= this.selectedReport.toMinute; 
+			this.reportForm.schedule= this.selectedReport.schedule; 
+			this.reportForm.schedulePeriod= this.selectedReport.schedulePeriod; 
+			this.reportForm.runDelayMinutes= this.selectedReport.runDelayMinutes; 
+            this.reportForm.scheduleCron= this.selectedReport.scheduleCron;//
+			this.reportForm.email= this.selectedReport.email; 
+			this.reportForm.includeData= this.selectedReport.includeData; 
+			this.reportForm.zipData= this.selectedReport.zipData; 
+            this.reportForm.recipients= this.selectedReport.recipients; 
+			
+			this.reportForm.points = this.selectedReport.points.map(x => {
+				const name = this.datapoints.find(p => p.pointId === x.id).name
+				return {...x, name}
+			});
+			if (copy)this.selectedId = -1
+			 
+		},
+		addEmail() {
+			if (!this.validateEmail(this.email)) return
+			if (this.reportForm.recipients.find(x => this.email === x.referenceAddress)) return  
+			this.reportForm.recipients.push({
+				referenceAddress: this.email,
+				referenceType: TYPE_ADDRESS
+			})
+			this.email = ''
+		},
+		addDatapoint({ pointId, name, colour = "", consolidatedChart = true }) {
+			if (this.reportForm.points.find(x => { return x.pointId === pointId})) return
+			this.reportForm.points
+				.push({ pointId, colour, consolidatedChart, name })
+		},
+		async fetchReportList() {
+			this.loading = true;
+			this.reportList = await this.$store.dispatch('fetchReports', {...this.options, keywords: this.search});
+			this.totalReports = this.reportList.total;
+		},
+		async fetchUserList() {
+			this.loading = true;
+			this.userList = [{ id: 0, username: "", email: ''}, ...(await this.$store.dispatch('getAllUsers'))];
+		},
+		runReport(id) {
+			this.$store.dispatch('runReport', id);
+			this.snackbar = true
+			this.snackbarMessage = `${this.$t('reportList.scriptExecuted')} `
+		},
+		updateDateRange() {
+			let bufferDate;
+			if (this.dateRange.endDate < this.dateRange.startDate) {
+				bufferDate = this.dateRange.endDate;
+				this.dateRange.endDate = this.dateRange.startDate;
+				this.dateRange.startDate = bufferDate;
+				this.dateRange.startTime = "00:00";
+				this.dateRange.endTime = "23:59";
+			}
+			if (this.dateRange.startDate) {
+				this.reportForm.fromYear=this.dateRange.startDate.split('-')[0]*1
```

Comment:
Nice solution for converting into the Number format! 

I have already knew this one:
```
this.reportForm.fromYear = +this.dateRange.startDate.split('-')[0]
```